### PR TITLE
Block Directory: Fix the block activation when metadata registered on server

### LINKED ISSUE #38245: Block Directory: Activation of a block with no title property will result in an error
### Description

Adding a block plugin from the block directory will cause an error under certain conditions.
It will probably occur for all blocks that do not specify **the title property** in the `registerBlockType` function.

This is also happening in the block plugin I publish, but the title property is listed in `block.json` and is not specified in `registerBlockType`.

It determines that there is no problem when I enter the slug of the plugin in Block Plugin Checker page, 

I think the checking rules should be unified for both `registerBlockType` and block directory.

Also, should I specify the title property in registerBlockType as well as in `block.json` to solve this error for the time being?

### Step-by-step reproduction instructions

1. Toggle bock inserter.
2. Enter the appropriate text in the search box.
3. Activate some of the blocks that were extracted in the block directory.
4. You should get a similar error if `registerBlockType` does not have a title property.

### Screenshots, screen recording, code snippet

issue

### Environment info

_No response_

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

### PR DESCRIPTION
**Pull request by gziolo**
## Description

Fixes #38245.

> ### Step-by-step reproduction instructions
> 
> 1. Toggle bock inserter.
> 2. Enter the appropriate text in the search box.
> 3. Activate some of the blocks that were extracted in the block directory.
> 4. You should get a similar error if `registerBlockType` does not have a title property.
> 
> ### Screenshots, screen recording, code snippet
> 
> issue

The action that installs the block from Block Directory triggers an additional REST API call to get the corresponding metadata for the newly installed plugin on the server. 

### Step-by-step reproduction instructions

1. Toggle bock inserter.
2. Enter the appropriate text (example: “list”) in the search box.
3. Activate some of the blocks that exist in the block directory, example: Super List.
4. You should see the block inserter correctly into the editor.

## Screenshots 

Screenshot 2022-02-10 at 11 39 49

https://user-images.githubusercontent.com/699132/153394689-79438243-db30-4b8d-a244-545b04f3b531.mov

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [-] My code follows the accessibility standards. 
- [-] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [-] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal). 
- [-] I've updated related schemas if appropriate.

### CODE REVIEW

**Comment by gziolo** [comment]
I see that an unit test is failing. I’m looking into it.

**Comment by gziolo** [comment]
I have added logic that covers also all those blocks that don't have PHP logic that register them on the server. Example block to use for testing:

Screenshot 2022-02-10 at 15 07 20

We need to account for 404 responses when the block is registered on the client. The good news is that client-side only blocks will work by design 😄

File: packages/block-directory/src/store/actions.js [hunk 1]
```diff
@@ -82,9 +85,44 @@ export const installBlockType = ( block ) => async ( {
...
+						'description',
+						'keywords',
+						'attributes',
+						'provides_context',
+						'uses_context',
+						'supports',
+						'styles',
+						'example',
+						'variations',
+					].reduce( ( accumulator, key ) => {
```
Comment (ryelle) [inline packages/block-directory/src/store/actions.js]: It took me a minute to parse that this is filtering out response properties - is there a reason we can't use the default `response`? or use `_fields` when requesting the API endpoint?

**Review by ryelle — APPROVED** [review]
I've tested a bunch of blocks and everything seems to be working great.

One minor comment about the `reduce` function, but if that's a common pattern I'm fine with it 🙂

File: packages/block-directory/src/store/actions.js (hunk 1 above)
Comment (gziolo) [reply packages/block-directory/src/store/actions.js]: I had a version with `_fields` before. I can switch back to it. Do you know how I also get rid of other meta fields like `_links`?

Comment (gziolo) [reply packages/block-directory/src/store/actions.js]: Addressed in https://github.com/WordPress/gutenberg/pull/38697/commits/e19bc491429e1364ed4db450d950b4361c27d3cd 👍🏻 

I noticed that the package already uses `loodash` so I went with `pick( response, metadataFields )` that should make it easier to read and removes `_links` field from the response.

**Pull request merged**